extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaBindingConverter.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.kafka.client.runtime;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+public class KafkaBindingConverter implements ServiceBindingConverter {
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        Optional<ServiceBinding> matchingByType = ServiceBinding.singleMatchingByType("kafka", serviceBindings);
+        if (!matchingByType.isPresent()) {
+            return Optional.empty();
+        }
+
+        Map<String, String> properties = new HashMap<>();
+        ServiceBinding binding = matchingByType.get();
+
+        String bootstrapServers = binding.getProperties().get("bootstrapServers");
+        if (bootstrapServers == null) {
+            bootstrapServers = binding.getProperties().get("bootstrap-servers");
+        }
+        if (bootstrapServers != null) {
+            properties.put("kafka.bootstrap.servers", bootstrapServers);
+        }
+
+        String securityProtocol = binding.getProperties().get("securityprotocol");
+        if (securityProtocol != null) {
+            properties.put("kafka.security.protocol", securityProtocol);
+        }
+
+        String saslMechanism = binding.getProperties().get("saslmechanism");
+        if (saslMechanism != null) {
+            properties.put("kafka.sasl.mechanism", saslMechanism);
+        }
+        String user = binding.getProperties().get("user");
+        String password = binding.getProperties().get("password");
+        if ("PLAIN".equals(saslMechanism) && (user != null) && (password != null)) {
```

Comment:
Is this supposed to be `PLAIN` or `PLAINTEXT`?

Reply:
https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_plain.html

```
`PLAIN` versus `PLAINTEXT` Do not confuse the SASL mechanism PLAIN with no SSL encryption being called PLAINTEXT. Configuration parameters such as sasl.enabled.mechanisms or sasl.mechanism.inter.broker.protocol may be configured to use SASL mechanism PLAIN, whereas security.inter.broker.protocol or listeners may be configured for no SSL encryption SASL_PLAINTEXT.
```

Reply:
This is what our operator is doing:
```
    this.saslMechanism = "PLAIN";
    this.securityProtocol = "SASL_SSL"; 
```